appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -2306,56 +2316,49 @@ private void closeApplication() {
   // Configure the system menu to include items to kill the application and to show "about"
   // information
 
+  public Menu getMenu() {
+    return menu;
+  }
+
   @Override
   public boolean onCreateOptionsMenu(Menu menu) {
-    // This procedure is called only once.  To change the items dynamically
-    // we would use onPrepareOptionsMenu.
-    super.onCreateOptionsMenu(menu);
-    // add the menu items
-    // Comment out the next line if we don't want the exit button
-    addExitButtonToMenu(menu);
-    addAboutInfoToMenu(menu);
-    for (OnCreateOptionsMenuListener onCreateOptionsMenuListener : onCreateOptionsMenuListeners) {
-      onCreateOptionsMenuListener.onCreateOptionsMenu(menu);
+    this.menu = menu;
+    Log.d(LOG_TAG, "onCreateOptionsMenu");
+    menu.add("dummy item");
+    for (OnCreateOptionsMenuListener listener : onCreateOptionsMenuListeners) {
+      listener.onCreateOptionsMenu(menu);
     }
     return true;
   }
 
-  public void addExitButtonToMenu(Menu menu) {
-    MenuItem stopApplicationItem = menu.add(Menu.NONE, Menu.NONE, Menu.FIRST,
-    "Stop this application")
-    .setOnMenuItemClickListener(new OnMenuItemClickListener() {
-      public boolean onMenuItemClick(MenuItem item) {
-        showExitApplicationNotification();
-        return true;
-      }
-    });
-    stopApplicationItem.setIcon(android.R.drawable.ic_notification_clear_all);
-  }
-
-  public void addAboutInfoToMenu(Menu menu) {
-    MenuItem aboutAppItem = menu.add(Menu.NONE, Menu.NONE, 2,
-    "About this application")
-    .setOnMenuItemClickListener(new OnMenuItemClickListener() {
-      public boolean onMenuItemClick(MenuItem item) {
-        showAboutApplicationNotification();
-        return true;
-      }
-    });
-    aboutAppItem.setIcon(android.R.drawable.sym_def_app_icon);
+  @Override
+  public boolean onPrepareOptionsMenu(Menu menu) {
+    menu.clear();
+    for (OnPrepareOptionsMenuListener listener : onPrepareOptionsMenuListeners) {
+      listener.onPrepareOptionsMenu(menu);
+    }
+    return true;
   }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-    for (OnOptionsItemSelectedListener onOptionsItemSelectedListener : onOptionsItemSelectedListeners) {
-      if (onOptionsItemSelectedListener.onOptionsItemSelected(item)) {
-        return true;
+    // handles the navigation drawer icon touch event
+    if(actionBarDrawerToggle.onOptionsItemSelected(item)) {
+      return true;
+    }
+    // handles the options menu item touch event
+    else {
```

Comment:
`else` should be on the line with the closing `}`. Move the comment to after the `{` or onto the next line.